src/components/v5/shared/UserInfoPopover/partials/UserNotVerified.tsx

```diff
@@ -1,22 +1,49 @@
 import React, { type FC } from 'react';
 
+import { Action } from '~constants/actions.ts';
+import { useActionSidebarContext } from '~context/ActionSidebarContext/ActionSidebarContext.ts';
 import { formatText } from '~utils/intl.ts';
+import { ACTION_TYPE_FIELD_NAME } from '~v5/common/ActionSidebar/consts.ts';
+import { ManageMembersType } from '~v5/common/ActionSidebar/partials/forms/ManageVerifiedMembersForm/consts.ts';
 import NotificationBanner from '~v5/shared/NotificationBanner/index.ts';
 
 const displayName = 'v5.UserInfoPopover.partials.UserNotVerified';
 
 export interface UserNotVerifiedProps {
   description?: React.ReactNode;
+  walletAddress: string;
 }
-// rename to UserNotVerified??
-const UserNotVerified: FC<UserNotVerifiedProps> = ({ description }) => {
+
+const UserNotVerified: FC<UserNotVerifiedProps> = ({
+  description,
+  walletAddress,
+}) => {
+  const {
+    actionSidebarToggle: [
+      ,
+      { toggleOn: toggleActionSidebarOn, toggleOff: toggleActionSidebarOff },
+    ],
+  } = useActionSidebarContext();
+
   return (
     <NotificationBanner
       status="warning"
       description={description}
       callToAction={
-        <button type="button">
-          {/* @TODO: add action */}
+        <button
+          type="button"
+          onClick={() => {
+            toggleActionSidebarOff();
+
+            setTimeout(() => {
+              toggleActionSidebarOn({
+                [ACTION_TYPE_FIELD_NAME]: Action.ManageVerifiedMembers,
+                members: [{ value: walletAddress }],
+                manageMembers: ManageMembersType.Add,
+              });
+            }, 500);
```

Comment:
Hey @CzarekDryl is the timeout necessary here? I'm new to the codebase so just trying to understand the reasoning behind it

Reply:
+1 on this question, maybe there was an issue here which could be solved with `requestAnimationFrame`?

Reply:
I think this will need reworking to remove the timeout, the 500ms delay makes this feel very unresponsive:

https://github.com/JoinColony/colonyCDapp/assets/34915414/fb1f38de-5868-450a-99aa-a905b9cdf1e5

With the timeout removed for comparison:

https://github.com/JoinColony/colonyCDapp/assets/34915414/e7246009-f50b-4202-9365-570f84e2b9b7

I assume this is purely to handle the scenario where the sidebar is already open and it needs to switch to a different action type? I imagine it would be okay for the form to instantly switch to another action without the sidebar closing and reopening. If the closing and reopening is a requirement, then maybe you can add a check here to see where it is being called from. Or might just need to add something to `toggleActionSidebarOff` to detect when the animation has finished / needs to run at all.

Reply:
@iamsamgibbs This should work better now, I've removed timeout when sidebar is closed.